src/django_components/component_registry.py

```diff
@@ -254,40 +265,24 @@ def settings(self) -> InternalRegistrySettings:
         """
         [Registry settings](../api#django_components.RegistrySettings) configured for this registry.
         """
-        # This is run on subsequent calls
-        if self._settings is not None:
-            # NOTE: Registry's settings can be a function, so we always take
-            # the latest value from Django's settings.
-            settings = self._settings()
-
-        # First-time initialization
         # NOTE: We allow the settings to be given as a getter function
         # so the settings can respond to changes.
-        # So we wrapp that in our getter, which assigns default values from the settings.
+        if callable(self._settings):
+            settings_input: Optional[RegistrySettings] = self._settings(self)
         else:
+            settings_input = self._settings
 
-            def get_settings() -> InternalRegistrySettings:
```

Comment:
I realized that this function was needlessly complex - Previously the function to return the registry settings was set to a variable on first access, and then that function was used directly to generate the settings.

Instead, the whole function can be regular method, which is called when needed.

I think I implemented the previous logic, but don't remember if there was a specific reason for that (And I don't think so).